code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -19,42 +18,29 @@
 	component_parts += new /obj/item/stock_parts/console_screen(null)
 	RefreshParts()
 
-/obj/machinery/chem_heater/upgraded/New()
-	..()
-	component_parts = list()
-	component_parts += new /obj/item/circuitboard/chem_heater(null)
-	component_parts += new /obj/item/stock_parts/micro_laser/ultra(null)
-	component_parts += new /obj/item/stock_parts/console_screen(null)
-	RefreshParts()
-
-/obj/machinery/chem_heater/RefreshParts()
-	heater_coefficient = 0.03
-	for(var/obj/item/stock_parts/micro_laser/M in component_parts)
-		heater_coefficient *= M.rating
-
 /obj/machinery/chem_heater/process()
 	..()
 	if(stat & NOPOWER)
 		return
-	var/state_change = 0
+	var/state_change = FALSE
 	if(on)
 		if(beaker)
-			if(beaker.reagents.chem_temp > desired_temp)
-				beaker.reagents.chem_temp += min(-1, max((desired_temp - beaker.reagents.chem_temp) * heater_coefficient, -15))
-			if(beaker.reagents.chem_temp < desired_temp)
-				beaker.reagents.chem_temp += max(1, min((desired_temp - beaker.reagents.chem_temp) * heater_coefficient, 15))
-			beaker.reagents.chem_temp = round(beaker.reagents.chem_temp) //stops stuff like 456.12312312302
-
-			beaker.reagents.handle_reactions()
-			state_change = 1
+			if(!beaker.reagents.total_volume)
+				on = FALSE
+				SSnanoui.update_uis(src)
+				return
+			beaker.reagents.temperature_reagents(desired_temp)
+			beaker.reagents.temperature_reagents(desired_temp)
```

Comment:
Ugly hack, really.

Goon's chem heater is a really weird infinite loop (albeit controlled). That operates on a `spawn`. Their heater doesn't actually `process`. Either case, it heats every second; our objects process every 2 seconds, so I just had it run this twice, to compensate.

Reply:
Creativity